adobecom/milo
# feat: added tags to lana logs

### Description
This PR addresses the requirement of enabling filtering capabilities for Splunk logs based on tags from LANA logs. 

Recommended tags:
`info` - network issues
`warn` - authoring related misconfigs or similar
`error` - actual error ( ex. cannot read Y of undefind )
`<module>`- extra tag to help centralise all logs related to a specific module

### Related Issue
Resolves: [MWPW-138031](https://jira.corp.adobe.com/browse/MWPW-138031)

### Testing instructions
1. Change the index value in the Splunk search to `lana_nonprod`
2. Add a `tags` key with an array of strings as value to the json sent as argument for the `lanaLog` method 
or if you're using the `logErrorFor` method add the array of strings as the third argument ( the strings in the array are the tags that will be associated with the log )
E.g.
`lanaLog({ message: 'Could not create global navigation.', e,  tags: 'errorType=error,module=gnav'  });`
`logErrorFor(onToggleClick, 'Toggle click failed', 'errorType=error,module=gnav'  ));`
3. All the logs sent from your local environment should show in Splunk 


## Screenshots:
<img width="1724" alt="Screenshot 2023-10-27 at 16 31 50" src="https://github.com/adobecom/milo/assets/146744221/09f6624d-3ef7-48af-be13-33801a6c6136">


## Test URLs
**Acrobat:**
- Before: https://main--dc--adobecom.hlx.page/drafts/rbogos/default?martech=off
- After: https://main--dc--adobecom.hlx.page/drafts/rbogos/default?milolibs=mwpw-138031-lana-logs-tags--milo--rbogos&martech=off

**BACOM:**
- Before: https://main--bacom--adobecom.hlx.page/drafts/rbogos/default
- After: https://main--bacom--adobecom.hlx.page/drafts/rbogos/default?milolibs=mwpw-138031-lana-logs-tags--milo--rbogos&martech=off

**CC:**
- Before: https://main--cc--adobecom.hlx.page/drafts/rbogos/default?martech=off
- After: https://main--cc--adobecom.hlx.page/drafts/rbogos/default?milolibs=mwpw-138031-lana-logs-tags--milo--rbogos&martech=off

**Milo:**
- Before: https://main--milo--robert-bogos.hlx.page/drafts/rbogos/default?martech=off
- After: https://lana-logs-tags--milo--robert-bogos.hlx.page/drafts/rbogos/default?martech=off
